misc/python3/ratchet.py

```diff
@@ -0,0 +1,377 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+'''Ratchet up passing tests, or ratchet down failing tests.
+
```

Comment:
This could use another sentence or so explaining what this actually does (i.e. what does "ratchet" mean in this context?) I *think* it means that we add more tests to some "should pass" list if they're already passing? But also I read through all the code to think that :)

Reply:
Fair enough, also re:

> I think the "ratchet" helper could use some more docstrings for some of the utility methods in there that aren't documented (that said, I am not familiar with doctest).

@wearpants has asked me to evaluate [`pytest-quarantine`](https://github.com/EnergySage/pytest-quarantine) as an alternative to `ratchet.py`, since it's got more community momentum (or at least more packaging :D). Agenda item for Thursday's call?

Reply:
To be clear: I'm planning to focus on `pytest-quarantine` between now and Thursday's call, to inform our discussion. If we decide to move forward with `ratchet.py` then yes, let's document it further.

Reply:
Sure! ("Less custom code" is good, in general ;)

Perhaps also writing a ticket on Trac describing what the goal is would be good (you appear to have an account now, so that's good!) ... then we could merge this PR without that part (and do ratchet, or pytest-quarantine or whatever else later).

Reply:
Ftr, decided on call last Thursday to stick with ratchet in order to avoid porting the unit test suite to pytest.